src/main.rs

```diff
@@ -193,10 +194,28 @@ fn main() -> Result<(), Box<dyn Error>> {
             verbose,
             union_as_struct,
         } => {
-            let file = std::fs::File::open(&file)?;
-            let file = unsafe { memmap::Mmap::map(&file) }?;
-            let file = object::File::parse(&*file)?;
-            let btf = Btf::load(&file)?;
+            let mut file = std::fs::File::open(&file)?;
+
+            // Read the magic number first.
+            let mut buffer = [0u8; 2];
+            file.read_exact(&mut buffer)?;
+            let magic_number: u16 = u16::from_le_bytes(buffer);
+
+            // And then the full file content.
+            let mut contents = Vec::with_capacity(usize::try_from(file.metadata()?.len())?);
```

Comment:
can you add BTF::load_file() API which would detect raw vs ELF internally and just return BTF? And then use that at least for Stat command below as well. Both dump and state are frequently used (by me, at least), so would be good to keep them both in sync in terms of what they support.